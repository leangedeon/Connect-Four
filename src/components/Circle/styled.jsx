import styled from 'styled-components'

export const CircleDiv = styled.div`
    height:70px;
    width:70px;
    background:#E76AFA;
    -moz-border-radius:50px;
    -webkit-border-radius:50px;
    border-radius:50px;
    background: ${(props) => {
            if (props.color) {
                if (props.color === 1) {
                    return '#E76AFA';
                }else{
                    return '#E76000';
                }
            }else{
                return "white";
            }
        }
    };
`;
