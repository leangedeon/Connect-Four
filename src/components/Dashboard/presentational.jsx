import React, { useEffect } from 'react';
import PropTypes from "prop-types";
import { Row } from "../../components/Row/index.jsx";
import { ListCoins } from "../../components/ListCoins/index.jsx";
import { reverseObject } from '../../utils';
import { Notification } from '../Notification/index.jsx';
import { Status } from '../Status/index.jsx';
const PATH_API = 'http://localhost:3001';

export const Dashboard = ({game, gameId, player, setPlayer, loading, setGame}) => {
    const gameReversed = reverseObject(game.dashboard);

    const createGame = async () => {
        const options = {  headers: { 'Content-Type': 'application/json' }, method: 'POST'};
        const data = await fetch(`${PATH_API}/game`, options);
        const dataJSON = await data.json();
        return dataJSON;
    }

    const getGame = async () => {
        const data = await fetch(`${PATH_API}/game`)
        const dataJSON = await data.json()
        return dataJSON;
    }

    useEffect( async () => {

        if (!gameId) {
            let game = {};

            if(player === 1)  {
                game = await createGame();
            } else {
                game = await getGame();
            }

            setGame(game);
            setPlayer(player);
        }
    }, []);

    return (
        <div>
            <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
                <div className="container-fuild">
                    <a href="/" className="navbar-brand">
                        Connect Four
                    </a>
                </div>
            </nav>
            <main className="bg">
                <div className="container">
                    <div className="row">
                        <h3>Welcome player #{player}</h3>
                        <p className="text-secondary">Game Id {gameId}</p>
                    </div>
                    <Status gameState={game.state}/>
                    
                    <div>

                        {
                            (loading) ? <Notification /> : <ListCoins quantity={6} />
                        }

                        {
                            gameReversed.map((row, index) => (
                                <Row rowData={row} key={`row_${index}`} />
                            ))
                        }
                    </div>
                </div>
            </main>
        </div>
    )

}

Dashboard.propTypes = {
    player: PropTypes.number.isRequired,
    loading: PropTypes.bool.isRequired,
    setPlayer: PropTypes.func.isRequired,
    gameId: PropTypes.string,
    setGame: PropTypes.func.isRequired,    
    game: PropTypes.shape({
        dashboard: PropTypes.object,
        state: PropTypes.string
      })
}
