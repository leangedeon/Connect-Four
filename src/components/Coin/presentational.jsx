import React from "react";
import PropTypes from "prop-types";
import { StyledButton } from './styled.jsx';
const PATH_API = process.env.PATH_API;

export const Coin = ({position, gameId, setLoading, setGame, player}) => {

    const persistMovement = async (gameId, position, player) => {
        const options = {  headers: { 'Content-Type': 'application/json' }, method: 'POST', body: JSON.stringify({'player': player})};
        const data = await fetch(`${PATH_API}/game/${gameId}/move/${position}`, options);
        const dataJSON = await data.json();
        return dataJSON;
    }

    const getGame = async (gameId) => {
        const data = await fetch(`${PATH_API}/game/${gameId}`)
        const dataJSON = await data.json()

        return dataJSON;
    }

    const checkTurn = async (gameId, player) => {
        const data = await getGame(gameId);

        if (data.next !== player && data.state !== 'DONE') {
            return checkTurn(gameId, player);
        }
    }

    const makeMovement = async (gameId, position, player) => {
        setLoading(true);

        const updatedGame = await persistMovement(gameId, position, player);

        setGame(updatedGame);

        await checkTurn(gameId, player);
        setLoading(false);

        const updatedGameAfterPlay = await getGame(gameId);

        setGame(updatedGameAfterPlay);
    }

    return (
        <StyledButton onClick={() => makeMovement(gameId, position, player)}>
            <button type="button" className="btn btn-primary">Put Coin</button>
        </StyledButton>
    )
}

Coin.propTypes = {
    position: PropTypes.number,
    gameId: PropTypes.string,
    setGame: PropTypes.func,
    setLoading: PropTypes.func,
    player: PropTypes.number
}


