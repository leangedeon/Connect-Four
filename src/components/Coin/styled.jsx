import styled from 'styled-components'

export const StyledButton = styled.div`
    margin: 10px 8px 10px 0;
    width: 70px;
    display: inline-block;
    cursor: pointer;
`;
