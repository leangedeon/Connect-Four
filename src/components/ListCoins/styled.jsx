import styled from 'styled-components'

export const StyledList = styled.div`
    max-width: 470px;
    height: 80px;
    margin: 0 auto;
`;