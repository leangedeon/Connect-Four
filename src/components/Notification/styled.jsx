import styled from 'styled-components'

export const StyledButton = styled.div`
    max-width: 485px;
    text-align: center;
    margin: 0 auto;
    height: 80px;
`;
