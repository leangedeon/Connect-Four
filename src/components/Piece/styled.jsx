import styled from 'styled-components'

export const StyledPiece = styled.div`
    display: inline-table;
    max-width: 16.5%;
    height: 80px;
    padding: 10px;
`;
