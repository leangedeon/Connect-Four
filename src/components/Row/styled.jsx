import styled from 'styled-components'

export const StyledRow = styled.div`
    max-width: 600px;
`;
