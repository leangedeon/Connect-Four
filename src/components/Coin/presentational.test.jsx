import React from 'react';
import { shallow } from 'enzyme';
import { Coin } from './presentational';
import { StyledButton } from './styled.jsx';


describe('Coin', () => {
    let props = {};
    let wrapper = {};
    const makeMovement = jest.fn();

    beforeEach(()=> {    
        props = {
            "position": 1,
            "gameId": 'some_game',
            "player": 1,
            "setGame": jest.fn(),
            "setLoading": jest.fn()
        };

        wrapper = shallow(<Coin {...props} />);
    });

    it('should match snapshot', () => {

        expect(wrapper).toMatchSnapshot();
    });

    it('should call some method when i click it', () => {

        const component = wrapper.find(StyledButton);

        component.simulate('click');
        expect(makeMovement).toHaveBeenCalled();
    });
});