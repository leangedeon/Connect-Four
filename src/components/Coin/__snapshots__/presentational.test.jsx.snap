// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin should match snapshot 1`] = `
<styled.div
  onClick={[Function]}
>
  <button
    className="btn btn-primary"
    type="button"
  >
    Put Coin
  </button>
</styled.div>
`;
